### PR TITLE
bugfix: failure to compete for a lock causes the global transaction to exit

### LINKED ISSUE #3442: 锁竞争失败导致rm退出全局事务
- [x] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
      Set auto-commit to true when local transactions are not being used. Failure to compete for a lock causes the global transaction to exit, invaliding the global row lock and dirty writing of the data.

      当不使用本地事务时，设置自动提交为true，锁竞争失败会导致退出全局事务，使得全局行锁失效，数据被脏写

### Ⅱ. Describe what happened

   脱离seata事务控制，全局行锁失效，数据被脏写。

```
Just paste your stack trace here!
```

### Ⅲ. Describe what you expected to happen

     锁竞争失败，不退出全局事务

### Ⅵ. Environment:

- JDK version :jdk1.8
- OS : macOS 15
- Others:

### PR DESCRIPTION
**Pull request by litianyu1992**
Ⅰ. Describe what this PR did

bugfix

Ⅱ. Does this pull request fix one issue?

fix #3442 

Ⅲ. Why don't you add test cases (unit test/integration test)?

Ⅳ. Describe how to verify it

可以下载该代码，https://github.com/ben905713861/seataTest，去除本地事务注解，并发访问测试

Ⅴ. Special notes for reviews

### CODE REVIEW

**Review by a364176773 — APPROVED** [review]
LGTM

**Review by caohdgege — APPROVED** [review]
LGTM

File: changes/en-us/1.5.0.md [hunk 1]
```diff
@@ -23,6 +23,7 @@
   - [[#3348](https://github.com/seata/seata/pull/3348)] support redis sentinel mode	
   - [[#2667](https://github.com/seata/seata/pull/2667)] support password decryption	
   - [[#3427](https://github.com/seata/seata/pull/3427)] add distributed lock interface	
+  - [[#3443](https://github.com/seata/seata/pull/3443)] send the `seata-server` log to `logstash` or `kafka`
```
Comment (l81893521) [inline changes/en-us/1.5.0.md:26]: Is this duplicated?

Comment (litianyu1992) [reply changes/en-us/1.5.0.md:26]: #3443属于feature,但是他填写到了bugfix,所以在我的pr里，帮他做了更正

Comment (litianyu1992) [reply changes/en-us/1.5.0.md:26]: > Is this duplicated?
#3443属于feature,但是他填写到了bugfix,所以在我的pr里，帮他做了更正

Comment (litianyu1992) [reply changes/en-us/1.5.0.md:26]: > Is this duplicated?

#3443属于feature,但是他填写到了bugfix,所以在我的pr里，帮他做了更正 @l81893521

**Review by l81893521 — APPROVED** [review]
LGTM

**Review by xingfudeshi — APPROVED** [review]
LGTM.

**Pull request merged**